aptos-move/move-examples/metapixel/sources/catcha_token.move

```diff
@@ -0,0 +1,292 @@
+module metapixel::catcha_token {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::string::{Self, String};
+    use std::signer;
+
+    use aptos_framework::object::{Self, ConstructorRef, Object};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::property_map;
+    use aptos_token_objects::royalty;
+    use aptos_token_objects::token;
+
+    // The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+    // The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 6;
+
+
+    struct CatchaCollection has key {
+        /// Used to mutate collection fields
+        mutator_ref: Option<collection::MutatorRef>,
+        /// Used to mutate royalties
+        royalty_mutator_ref: Option<royalty::MutatorRef>,
+        /// Determines if the creator can mutate the collection's description
+        mutable_description: bool,
```

Comment:
I think we might not need all of these since the creators of Catcha would want all of these powers always